client/vaultclient/vaultclient.go

```diff
@@ -285,7 +286,7 @@ func (c *vaultClient) DeriveToken(alloc *structs.Allocation, taskNames []string)
 }
 
 // DeriveTokenWithJWT returns a Vault ACL token using the JWT login endpoint.
-func (c *vaultClient) DeriveTokenWithJWT(ctx context.Context, req JWTLoginRequest) (string, error) {
+func (c *vaultClient) DeriveTokenWithJWT(ctx context.Context, req JWTLoginRequest, ns string) (string, error) {
```

Comment:
Thoughts on having the namespace as part of `JWTLoginRequest`? It feels like the namespace is an important part of the request?

Reply:
Oh, that would be nicer. Will do.